File: Domain/Experience.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Experience
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Category { get; set; }
        public DateTime Date { get; set; }
        public string Main { get; set; }
        public string Main2 { get; set; }
        public string City { get; set; }
        public string Image { get; set; }
        public string skills { get; set; }
        public string Link1 { get; set; }
        public string Link1Name { get; set; }
        public string Link2 { get; set; }
        public string Link2Name { get; set; }
        // public string Image { get; set; }
        public DateTime DateStarted { get; set; }
        public DateTime DateEnded { get; set; }
        public virtual ICollection<UserExperience> UserExperience {get; set;}
        // public virtual ICollection<Comment> Comments {get; set;}
    }
}

File: client-app/src/App/Models/experience.ts

export interface IExperiencesEnvelope {
    experiences: IExperience[];
    experienceCount: number;
  }

export interface IExperience {
    id: string;
    title: string;
    main: string;
    main2: string;
    category: string;
    date: Date;
    dateStarted: Date;
    dateEnded: Date;
    isHost: boolean;
    isLiked: boolean;
    liked: IAttendee[];
    comments: IComment[];
    link1: string;
    link1Name: string;
    link2: string;
    link2Name: string;
    skills: string;
    image: string;
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
  }

export interface IExperienceFormValues extends Partial<IExperience> {
    time?: Date;
}
export class experienceFormValues implements IExperienceFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = "";
    description: string = "";
    date?: Date = undefined;
    time?: Date = undefined;
    city?: string = '';
    image?: string = '';
    main2?: string = '';
    main?: string = '';
    link1?: string = '';
    link1Name?: string = '';
    link2?: string = '';
    link2Name?: string = '';
    skills?: string = '';
    dateEnded?: Date = undefined;
    dateStarted?: Date = undefined;

    constructor(init?: IExperienceFormValues) {
        if (init && init.date){
            init.time = init.date
        }
        Object.assign(this, init);
    } 
}

export interface IAttendee {
    username: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean;
}